policy/modules/services/incus.te

```diff
@@ -0,0 +1,209 @@
+policy_module(incus)
+########################################
+#
+# Declarations
+#
+
+# common attribute for all container domains
+# that may be used with incus
+attribute incus_container_domain;
+
+container_engine_domain_template(incusd)
+container_system_engine(incusd_t)
+type incusd_exec_t;
+
+container_engine_executable_file(incusd_exec_t)
+
+init_daemon_domain(incusd_t, incusd_exec_t)
+
+ifdef(`enable_mls',`
+	init_ranged_daemon_domain(incusd_t, incusd_exec_t, s0 - mls_systemhigh)
+')
+
+type incusc_t;
+type incusc_exec_t;
+
+container_engine_executable_file(incusc_exec_t)
+application_domain(incusc_t, incusc_exec_t)
+
+# other engines like docker have this, what will it do?
+#container_engine_domain_template(incusd_user)
+
+# user containers are not tested yet
+#container_user_engine(incusd_user_t)
+#application_domain(incusd_user_t, incusd_exec_t)
+
+#mls_trusted_object(incusd_user_t)
+
+type incusc_user_t;
+application_domain(incusc_user_t, incusc_exec_t)
+
+########################################
+#
+# incus daemon local policy
+
+allow incusd_t self:process getattr;
+allow incusd_t self:anon_inode { create map read write };
+allow incusd_t self:io_uring sqpoll;
+allow incusd_t self:netlink_kobject_uevent_socket create_socket_perms;
+allow incusd_t self:vsock_socket create_stream_socket_perms;
+dontaudit incusd_t self:capability sys_module;
+
+# incusd wants to load netdev-phys*
+kernel_request_load_module(incusd_t)
+
+ifdef(`init_systemd',`
+	init_start_system(incusd_t)
+	init_stop_system(incusd_t)
+	init_get_system_status(incusd_t)
+	init_stop_generic_units(incusd_t)
+')
+
+# losetup required for instance creation
+dev_rw_loop_control(incusd_t)
+dev_rw_vhost(incusd_t)
+
+# comm="incusd" path="/run/systemd/resolve/stub-resolv.conf"
+container_read_tmpfs_files(incusd_t)
+
+fs_getattr_binfmt_misc_fs(incusd_t)
+fs_mount_binfmt_misc_fs(incusd_t)
+fs_getattr_binfmt_misc_dirs(incusd_t)
+fs_dontaudit_manage_fusefs_files(incusd_t)
+
+# here dontaudit would also work, but consuming more resources
+fs_watch_tmpfs_dirs(incusd_t)
+
+dnsmasq_domtrans(incusd_t)
+dnsmasq_signal(incusd_t)
+dnsmasq_kill(incusd_t)
+dnsmasq_signull(incusd_t)
+
+kernel_setsched(incusd_t)
+dpkg_dontaudit_manage_db(incusd_t)
+
+files_exec_usr_files(incusd_t)
+
+iptables_domtrans(incusd_t)
+iptables_kill(incusd_t)
+
+mount_exec(incusd_t)
+
+rsync_exec(incusd_t)
+
+
+# Error: Failed pre-start check for device "eth0": Network "incusbr0" unavailable on this server
+allow incusd_t self:rawip_socket { create getopt ioctl read setopt write };
+
+# Error: Failed instance creation: Failed to start device "eth0":
+#  Failed to create the tap interfaces "tape73a8425":
+#  Failed to run: ip tuntap add name tape73a8425 mode tap multi_queue:
+#  exit status 1 (ioctl(TUNSETIFF): Permission denied)
+allow incusd_t self:tun_socket { create relabelfrom relabelto };
```

Comment:
These can go up with the other `allow incusd_t self:*` rules with a (optional) consolidated comment that incusd will manage the host network.